src/sidebar/directive/group-list.js

```diff
@@ -20,6 +20,10 @@ function GroupListController($scope, $window, groups, settings) {
   $scope.focusGroup = function (groupId) {
     groups.focus(groupId);
   };
+
+  if (settings.services && settings.services[0].icon) {
```

Comment:
We have a bunch of checks for `settings.services` in several places that are all different. I'd suggest refactoring that into a helper. eg:

```js
// In some helper module...
function annotationService(settings) {
  if (!Array.isArray(settings.services) || settings.services.length === 0) {
    return null;
  }
  return settings.services[0];
}

// In places that use the service
var svc = annotationService(settings);
if (svc && svc.icon) { ... }
```